homework/index.js

```diff
@@ -30,18 +34,63 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+  function createLayout() {
+    const root = document.getElementById('root');
+    createAndAppend('select', root, { id: 'repo-select' });
+    createAndAppend('option', document.getElementById('repo-select'), { text: 'Choose a Repository' });
+    createAndAppend('div', root, { id: 'body-container' });
+    createAndAppend('div', document.getElementById('body-container'), { id: 'repo-details' });
+    createAndAppend('div', document.getElementById('body-container'), { id: 'contributors' });
+
+  }
+
+  function getRepoDataFromOrgAndAddToDOM() {
+    const REPOS_URL = 'https://api.github.com/orgs/foocoding/repos?per_page=100';
+    fetchJSON(REPOS_URL).then(function (response) {
+      response.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase()) ? 1 : -1);
+      response.forEach(repoDataObj => {
+        createAndAppend('option', document.getElementById('repo-select'), { text: repoDataObj.name })
+      })
+      addListenerOnSelect(response);
+    })
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
+  function addDataToRepoDetails(data) {
+    const repoDetailsDiv = document.getElementById('repo-details');
+    repoDetailsDiv.innerHTML = '';
+    createAndAppend('div', repoDetailsDiv, { id: "repoNameID" });
+    createAndAppend('a', document.getElementById("repoNameID"), { href: data.html_url, target: "_blank", text: data.name })
+    createAndAppend('div', repoDetailsDiv, { text: `Repository Description: ${data.description}` });
+    createAndAppend('div', repoDetailsDiv, { text: `Number of Forks: ${data.forks}` });
+  }
+
+  function getContributors(data) {
+    const contributorsDiv = document.getElementById('contributors');
+    contributorsDiv.innerHTML = 'Contributors:';
+    fetchJSON(data.contributors_url).then(function (response) {
```

Comment:
I miss the displaying of errors to the user, have a look at `catch` and see if you can figure it out!